File: 2019/python/day02.py
for noun in range(100):
    for verb in range(100):
        input = list(map(int,open("../input/day2.input").readline().split(",")))
        ip, opcode, input[1], input[2] = 0, input[0], noun, verb
        while opcode != 99:
            if opcode == 1: input[input[ip + 3]] = input[input[ip + 1]] + input[input[ip + 2]]
            if opcode == 2: input[input[ip + 3]] = input[input[ip + 1]] * input[input[ip + 2]]
            ip = ip + 4
            opcode = input[ip]
        if (noun == 12 and verb == 2): print ("answer a: {}".format(input[0]))
        if input[0] == 19690720: print ("answer b: {}".format(100 * noun + verb))

File: 2019/golang/day02.go
package main
import (
    "fmt"
    "io/ioutil"
    "strings"
    "strconv"
)

func main() {
    for noun := 0; noun < 100; noun++ {
        for verb := 0; verb < 100; verb++ {
            values, _ := ioutil.ReadFile("../input/day2.input")
            input := []int{}
            for _, s := range strings.Split(string(values), ",") {
                i, _ := strconv.Atoi(s)
                input = append(input, i)
            }
            ip, opcode := 0, input[0]
            input[1], input[2] = noun, verb
            for opcode != 99 {
                if opcode == 1 {
                    input[input[ip + 3]] = input[input[ip + 1]] + input[input[ip + 2]]
                }
                if opcode == 2 {
                    input[input[ip + 3]] = input[input[ip + 1]] * input[input[ip + 2]]
                }
                ip += 4
                opcode = input[ip]
            }
            if noun == 12 && verb == 2 {
                fmt.Printf("answer a: %v\n", input[0])
            }
            if input[0] == 19690720 {
                fmt.Printf("answer b: %v\n", 100 * noun + verb)
            }
        }
    }
}
